radis/lbl/broadening.py

```diff
@@ -326,37 +326,42 @@ def pressure_broadening_HWHM(
     # | dev note: in that case we simplify the expression by calculation the
     # | power function once only.
 
-    diluent_molecules = diluent.keys()
-
-    # check if gamma_diluent and n_diluent exists or not
-    try:
-        gamma_lb = 0
-        for diluent_molecule, diluent_mole_fraction in diluent.items():
-            if diluent_molecule != "air":
-                gamma_lb += (
-                    (Tref / Tgas)
-                    ** diluent_broadening_coeff["n_" + diluent_molecule.lower()]
-                ) * (
-                    diluent_broadening_coeff["gamma_" + diluent_molecule.lower()]
-                    * pressure_atm
-                    * diluent_mole_fraction
+    # check if gamma_diluent and n_diluent exists or not and in case they are not present use broadening coefficient of air instead
+    gamma_lb = 0
+    for diluent_molecule, diluent_mole_fraction in diluent.items():
+            try:
+                    gamma_lb += (
+                        (Tref / Tgas)
+                        ** diluent_broadening_coeff["n_" + diluent_molecule.lower()]
+                    ) * (
+                        diluent_broadening_coeff["gamma_" + diluent_molecule.lower()]
+                        * pressure_atm
+                        * diluent_mole_fraction
+                    )
+            except KeyError:
+                # import warnings
+                # from radis.misc.warning import AccuracyWarning
+                # if diluent_molecule != "air":
+                #     warnings.warn(
+                #         AccuracyWarning(
+                #             "Broadening Coefficient of "+diluent_molecule+" not present in database using broadening coefficient of air instead. \n!!! Solution: Try using once `use_cached='regen'' in calc_spectrum!!!"
+                #         )
+                #     )
+                
+                ## A warning is normally raised already in '_calc_broadening_HWHM' when constructing the dict diluent_broadening_coef
+                gamma_lb += ((Tref / Tgas) ** Tdpair) * (
+                    (airbrd * pressure_atm * diluent["air"])
```

Comment:
[IMPORTANT] 
@1someshverma  this line introduces a strong physics error. 

- you removed air_broadening (or you expect that it is added when gamma_air is not found 
- if there is a missing value ; for instance gamma_CO2 is not found, it is replaced by airbrd with the mole fraction of air. This is **wrong**. We can use gamma_air/airbrd as a placeholder for gamma_CO2; BUT it should keep the mole_fraction of CO2; not air. 
 
This is a critical error ; that's ok because it didn't make it to the public version; but it should have been picked by our test suite. 

This means  : 
- we need to fix the error 
- we need to fix the test suite   @minouHub 
- also, try not using try/except statements in general.

Reply:
Test fixed by @erwanp in #589. @1someshverma, please replace `diluent["air"]` by `diluent_mole_fraction`